packages/react-ui-components/src/Dialog/DialogManager.ts

```diff
@@ -0,0 +1,52 @@
+export interface EventRoot {
+    addEventListener: Document['addEventListener'];
+    removeEventListener: Document['removeEventListener'];
+}
+
+export interface Dialog {
+    close: () => void;
+}
+
+export class DialogManager {
+    private dialogs: Dialog[] = [];
+
+    constructor(private readonly deps: { eventRoot: EventRoot }) {}
+
+    public register(dialog: Dialog): void {
+        if (this.dialogs.length === 0) {
+            this.deps.eventRoot.addEventListener('keydown', this.handleKeydown);
+        }
+
+        if (!this.dialogs.includes(dialog)) {
+            this.dialogs.push(dialog);
+        }
+    }
+
+    public forget(dialog: Dialog): void {
+        this.dialogs = this.dialogs.filter((d) => d !== dialog);
+        this.removeHandleKeydownEventListenerIfNecessary();
+    }
+
+    public readonly handleKeydown = (event: KeyboardEvent): void => {
+        if (event.key === 'Escape') {
+            this.closeLatest();
+        }
+    }
+
+    public closeLatest(): void {
+        const dialog = this.dialogs.pop();
+        if (dialog) {
+            dialog.close();
+            this.removeHandleKeydownEventListenerIfNecessary();
+        }
```

Comment:
```suggestion
        const lastDialog = this.dialogs[this.dialogs.length - 1];
        if (lastDialog) {
            this.forget(lastDialog);
        }
```

as `closeLatest` is only used for testing, id rather have it cover the `forget` method aswell ^^